src/components/AppIcon.jsx

```diff
@@ -1,30 +1,81 @@
 import React from 'react'
 import { translate } from 'cozy-ui/react/I18n'
+import defaultIcon from '../assets/icons/16/icon-cube-16.svg'
+import ComingSoonModal from 'components/ComingSoonModal'
+import { appShape } from '../proptypes'
 
-export const AppIcon = ({ t, label, dataIcon, iconSrc, href, comingSoon = false, blurry = false, toggle }) => {
-  if (href) toggle = null // href always have priority
-  let appClass = comingSoon ? 'coz-bar-coming-soon-app' : ''
-  if (toggle) {
-    appClass += ' --toggable'
-    href = '#'
+const HAS_COMING_SOON_DESCRIPTION = {
+  store: true
+}
+
+class AppIcon extends React.Component {
+  state = {
+    showingComingSoon: false
+  }
+
+  toggleComingSoon = (showing) => {
+    this.setState({
+      showingComingSoon: showing
+    })
+  }
+
+  openComingSoon = (ev) => {
+    this.toggleComingSoon(true)
+    if (ev) {
+      ev.preventDefault()
+    }
+  }
+
+  closeComingSoon = () => {
+    this.toggleComingSoon(false)
   }
-  return (
-    <li className='coz-nav-item'>
-      <a role='menuitem' href={href} data-icon={dataIcon} className={appClass} title={label} onClick={toggle}>
-        {iconSrc &&
-          <img src={iconSrc} alt='' width='64' height='64' className={blurry && 'blurry'} />
-        }
-        {comingSoon &&
-          <span
-            className='coz-bar-coming-soon-badge'
-          >
-            {t('soon')}
-          </span>
-        }
-        <p className='coz-label'>{label}</p>
-      </a>
-    </li>
-  )
+
+  render () {
+    const { t, app } = this.props
+    const dataIcon = app.icon ? `icon-${app.slug}` : ''
+    const blurry = !app.icon || !app.icon.cached || false
+    const label = (app.namePrefix ? (app.namePrefix + ' ') : '') + app.name
+    const comingSoon = app.comingSoon || false
+    const iconSrc = app.icon && app.icon.cached ? app.icon.src : defaultIcon
+    const canShowComingSoonDescription = HAS_COMING_SOON_DESCRIPTION[app.slug]
```

Comment:
I would use `hasOwnProperty` here to avoid getting true with a property from prototype but it's clearly not blocking